File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1

        while True:
            m = l + int((r - l) / 2)

            num = nums[m]

            if num == target:
                return m

            if l == r:
                break

            if num < target:
                l = m + 1
            else:
                r = m

        return -1


        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;
        for (;;) {
            int s = numbers[l] + numbers[r];
            if (s > target) {
                r--;
            } else if (s < target) {
                l++;
            } else if (l != r) {
                return new int[]{l + 1, r + 1};
            }
        }
    }
}
